src/components/AttachmentModal.tsx

```diff
@@ -1,186 +1,218 @@
+import {isNotEmptyObject} from '@src/types/utils/EmptyObject';
 import AttachmentCarousel from './Attachments/AttachmentCarousel';
 import AttachmentView from './Attachments/AttachmentView';
 import Button from './Button';
 import ConfirmModal from './ConfirmModal';
 import HeaderGap from './HeaderGap';
 import HeaderWithBackButton from './HeaderWithBackButton';
 import * as Expensicons from './Icon/Expensicons';
-import sourcePropTypes from './Image/sourcePropTypes';
 import Modal from './Modal';
 import SafeAreaConsumer from './SafeAreaConsumer';
-import transactionPropTypes from './transactionPropTypes';
-import withLocalize, {withLocalizePropTypes} from './withLocalize';
-import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
 
 /**
  * Modal render prop component that exposes modal launching triggers that can be used
  * to display a full size image or PDF modally with optional confirmation button.
  */
 
-const propTypes = {
-    /** Optional source (URL, SVG function) for the image shown. If not passed in via props must be specified when modal is opened. */
-    source: PropTypes.oneOfType([PropTypes.string, sourcePropTypes]),
+type AttachmentModalOnyxProps = {
+    /** The transaction associated with the receipt attachment, if any */
+    transaction: OnyxEntry<OnyxTypes.Transaction>;
 
-    /** Optional callback to fire when we want to preview an image and approve it for use. */
-    onConfirm: PropTypes.func,
+    /** The report associated with the receipt attachment, if any */
+    parentReport: OnyxEntry<OnyxTypes.Report>;
 
-    /** Whether the modal should be open by default */
-    defaultOpen: PropTypes.bool,
+    /** The policy associated with the receipt attachment, if any */
+    policy: OnyxEntry<OnyxTypes.Policy>;
 
-    /** Optional callback to fire when we want to do something after modal show. */
-    onModalShow: PropTypes.func,
+    /** The list of report actions associated with the receipt attachment, if any */
+    parentReportActions: OnyxEntry<OnyxTypes.ReportActions>;
 
-    /** Optional callback to fire when we want to do something after modal hide. */
-    onModalHide: PropTypes.func,
+    /** Current user session */
+    // eslint-disable-next-line react/no-unused-prop-types
+    session: OnyxEntry<OnyxTypes.Session>;
+};
 
-    /** Optional callback to fire when we want to do something after attachment carousel changes. */
-    onCarouselAttachmentChange: PropTypes.func,
+type Attachment = {
+    source: string;
+    isAuthTokenRequired: boolean;
+    file: File;
+    isReceipt: boolean;
+};
 
-    /** Optional original filename when uploading */
-    originalFileName: PropTypes.string,
+type ImagePickerResponse = {
+    height: number;
+    name: string;
+    size: number;
+    type: string;
+    uri: string;
+    width: number;
+};
 
-    /** A function as a child to pass modal launching methods to */
-    children: PropTypes.func,
+type Data = File | ImagePickerResponse;
```

Comment:
`Data` sounds too common, maybe we can rename it to `FileObject` for example, since it's used to type same-naming variable in a couple of places